sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1844,3 +1844,34 @@ case class GpuStringToMap(strExpr: Expression,
     }
   }
 }
+
+case class GpuStringInstr(str: Expression, substr: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant  {
+  // Locate the position of the first occurrence of substr column in the given string.
+  // returns null if one of the arguments is null
+  // returns zero if not found
+  // return values are 1 based.
+  override def dataType: DataType = IntegerType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, StringType)
+  override def left: Expression = str
+  override def right: Expression = substr
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
```

Comment:
This case is supported in Spark.

Reply:
I just checked in cudf and see that this pair of parameter types is not yet supported. Since we are trying to match with Spark, we may fill a FEA in cudf to implement that.

Reply:
I believe this is not specific to `instr`, it is the case for all the string Functions.
I double checked that the documentation clearly states that the plugin only offer partial support for each Function.

Reply:
Having partial support doesn't mean we don't want to support. It's just because we can't yet (no cudf back API) or we forgot that (due to the given explicit example from the FEA issue) etc.

If we can, why don't try to cover all? We should not let the customer wait for their FEA then when the feature arrives they turn out to discover that it is only partially supported which doesn't meet their need.

Reply:
Yes, exactly that's what I meant.
We can file a single issue against CUDF and list all the operations that are currently dependent on this.